src/components/SingleRecipePage.tsx

```diff
@@ -0,0 +1,205 @@
+              transform="scale(-1,1)"
+              icon={<GiPencil />}
+              title="edit"
+            />
+            <IconButton
+              size="lg"
+              variant="outline"
+              aria-label="Add to menu planner"
+              icon={<GiCalendar />}
+              title="add to menu planner"
+            />
+            <IconButton
+              size="lg"
+              variant="outline"
+              aria-label="Add to shopping list"
+              icon={<GiShoppingCart />}
+              title="add to shopping cart"
+            />
+            <IconButton
+              size="lg"
+              variant="outline"
+              aria-label="Delete recipe"
+              icon={<IoTrashOutline />}
+              title="delete"
+            />
+            <IconButton
+              size="lg"
+              variant="outline"
+              aria-label="Print recipe"
+              icon={<TfiPrinter />}
+              title="print"
+            />
+          </Flex>
+        </GridItem>
+      </Grid>
+      <Grid templateColumns="repeat(3, 1fr)" gap={6}>
+        <GridItem colSpan={2} w="95%">
+          <Flex flexDirection="column">
+            <Box marginTop="5">
+              <Heading as="h3" size="md" marginBottom="3">
+                Ingredients
+              </Heading>
+              {recipe.recipeIngredients.map((ingredient, index) => (
+                <SingleRecipeIngredients key={index} ingredient={ingredient} />
+              ))}
+            </Box>
+            <Box marginTop="5">
+              <Heading as="h3" size="md" marginBottom="3">
+                Instructions
+              </Heading>
+              <Text>{recipe.recipeInstructions}</Text>
+            </Box>
+            <Box marginTop="5">
+              <Flex onClick={onToggle} cursor="pointer">
+                <Heading as="h3" size="md" marginBottom="3">
+                  Nutrition Information
+                </Heading>
+                <Box as="span">
+                  <Icon as={ChevronDownIcon} />
+                </Box>
+              </Flex>
+              <Collapse in={isOpen} animateOpacity>
+                {[
+                  {
+                    displayName: "Calories",
+                    content: recipe.recipeNutritionInfo.NutritionInfoCalories,
+                    unit: "kcal"
+                  },
+                  {
+                    displayName: "Carbohydrates",
+                    content: recipe.recipeNutritionInfo.NutritionInfoCarbs,
+                    unit: "g"
+                  },
+                  {
+                    displayName: "Protein",
+                    content: recipe.recipeNutritionInfo.NutritionInfoProtein,
+                    unit: "g"
+                  },
+                  {
+                    displayName: "Fat",
+                    content: recipe.recipeNutritionInfo.NutritionInfoFat,
+                    unit: "g"
+                  }
```

Comment:
NON-BLOCKING: I like this format, but for easier reading, you can store this array in a constant at the top of this component to make the `return` statement easier to read.

Reply:
Thank you! I will do it  in the branch I am working right now.